From e447a65fbeed7e61ee25cd7eb5bc123b9ad1cc26 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:41 -0400
Subject: [usb] usbtest: update bos test coverage to usb 2.1 device

Message-id: <1410887463-20674-210-git-send-email-dzickus@redhat.com>
Patchwork-id: 95301
O-Subject: [RHEL7 PATCH 209/591] usb: usbtest: update bos test coverage to usb 2.1 device
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit f625099f1a504742aef4c6ab20ba0ca981847e3c
Author: Huang Rui <ray.huang@amd.com>
Date:   Wed Nov 13 22:35:13 2013 +0800

    usb: usbtest: update bos test coverage to usb 2.1 device

    The commit "usb: usbtest: support bos descriptor test for usb 3.0"
    introduced a test for bos descriptor. And USB 2.1 device also can be
    checked. So this patch extends the test coverage to support USB 2.1 device.

    Reported-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Signed-off-by: Huang Rui <ray.huang@amd.com>
    Acked-by: Felipe Balbi <balbi@ti.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index 0b5c3b1..bff058e 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -763,7 +763,7 @@ static int ch9_postconfig(struct usbtest_dev *dev)
   * there's always [9.4.3] a bos device descriptor [9.6.2] in USB
   * 3.0 spec
   */
- if (le16_to_cpu(udev->descriptor.bcdUSB) >= 0x0300) {
+ if (le16_to_cpu(udev->descriptor.bcdUSB) >= 0x0210) {
   struct usb_bos_descriptor *bos = NULL;
   struct usb_dev_cap_header *header = NULL;
   unsigned total, num, length;
-- 
1.7.1